File: backend/src/main/java/com/lawfirm/domain/hr/entity/DevelopmentPlan.java
package com.lawfirm.domain.hr.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lawfirm.common.base.BaseEntity;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 个人发展规划实体
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("hr_development_plan")
public class DevelopmentPlan extends BaseEntity {

    /**
     * 规划编号
     */
    private String planNo;

    /**
     * 员工ID
     */
    private Long employeeId;

    /**
     * 员工姓名
     */
    private String employeeName;

    /**
     * 规划年度
     */
    private Integer planYear;

    /**
     * 规划标题
     */
    private String planTitle;

    /**
     * 当前职级ID
     */
    private Long currentLevelId;

    /**
     * 当前职级名称
     */
    private String currentLevelName;

    /**
     * 目标职级ID
     */
    private Long targetLevelId;

    /**
     * 目标职级名称
     */
    private String targetLevelName;

    /**
     * 目标达成日期
     */
    private LocalDate targetDate;

    /**
     * 职业目标（JSON数组）
     */
    private String careerGoals;

    /**
     * 技能目标（JSON数组）
     */
    private String skillGoals;

    /**
     * 业绩目标（JSON数组）
     */
    private String performanceGoals;

    /**
     * 行动计划（JSON数组）
     */
    private String actionPlans;

    /**
     * 所需培训
     */
    private String requiredTraining;

    /**
     * 所需资源
     */
    private String requiredResources;

    /**
     * 导师ID
     */
    private Long mentorId;

    /**
     * 导师姓名
     */
    private String mentorName;

    /**
     * 完成进度百分比
     */
    private Integer progressPercentage;

    /**
     * 进度备注
     */
    private String progressNotes;

    /**
     * 状态：DRAFT-草稿, ACTIVE-执行中, COMPLETED-已完成, CANCELLED-已取消
     */
    private String status;

    /**
     * 审核人ID
     */
    private Long reviewedBy;

    /**
     * 审核人姓名
     */
    private String reviewedByName;

    /**
     * 审核时间
     */
    private LocalDateTime reviewedAt;

    /**
     * 审核意见
     */
    private String reviewComment;
}

File: frontend/apps/web-antd/src/api/hr/development-plan.ts
/**
 * 员工发展计划 API
 */
import { requestClient } from '#/api/request';

import type { PageResult } from '../matter/types';

// ========== 类型定义 ==========
export interface DevelopmentMilestoneDTO {
  id: number;
  planId: number;
  milestoneName: string;
  description?: string;
  targetDate?: string;
  status?: string;
  statusName?: string;
  completedDate?: string;
  completionNote?: string;
  sortOrder?: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface DevelopmentPlanDTO {
  id: number;
  planNo?: string;
  employeeId: number;
  employeeName?: string;
  planYear: number;
  planTitle: string;
  currentLevelId?: number;
  currentLevelName?: string;
  targetLevelId?: number;
  targetLevelName?: string;
  targetDate?: string;
  careerGoals?: string[];
  skillGoals?: string[];
  performanceGoals?: string[];
  actionPlans?: string[];
  requiredTraining?: string;
  requiredResources?: string;
  mentorId?: number;
  mentorName?: string;
  progressPercentage?: number;
  progressNotes?: string;
  status?: string;
  statusName?: string;
  reviewedBy?: number;
  reviewedByName?: string;
  reviewedAt?: string;
  reviewComment?: string;
  milestones?: DevelopmentMilestoneDTO[];
  createdAt?: string;
  updatedAt?: string;
}

export interface DevelopmentPlanQuery {
  pageNum?: number;
  pageSize?: number;
  keyword?: string;
  status?: string;
  employeeId?: number;
  planYear?: number;
}

export interface CreateDevelopmentPlanCommand {
  planYear: number;
  planTitle: string;
  targetLevelId?: number;
  targetDate?: string;
  careerGoals?: string[];
  skillGoals?: string[];
  performanceGoals?: string[];
  actionPlans?: string[];
  requiredTraining?: string;
  requiredResources?: string;
  mentorId?: number;
  milestones?: MilestoneItem[];
}

export interface MilestoneItem {
  milestoneName: string;
  description?: string;
  targetDate?: string;
  sortOrder?: number;
}

// ========== API 函数 ==========

/** 分页查询发展规划 */
export function getDevelopmentPlanList(params: DevelopmentPlanQuery) {
  return requestClient.get<PageResult<DevelopmentPlanDTO>>('/hr/development-plan', { params });
}

/** 获取规划详情 */
export function getDevelopmentPlanDetail(id: number) {
  return requestClient.get<DevelopmentPlanDTO>(`/hr/development-plan/${id}`);
}

/** 获取我的当年规划 */
export function getMyCurrentDevelopmentPlan() {
  return requestClient.get<DevelopmentPlanDTO>('/hr/development-plan/my-current');
}

/** 创建发展规划 */
export function createDevelopmentPlan(data: CreateDevelopmentPlanCommand) {
  return requestClient.post<DevelopmentPlanDTO>('/hr/development-plan', data);
}

/** 更新发展规划 */
export function updateDevelopmentPlan(id: number, data: CreateDevelopmentPlanCommand) {
  return requestClient.put<DevelopmentPlanDTO>(`/hr/development-plan/${id}`, data);
}

/** 删除发展规划 */
export function deleteDevelopmentPlan(id: number) {
  return requestClient.delete(`/hr/development-plan/${id}`);
}

/** 提交规划 */
export function submitDevelopmentPlan(id: number) {
  return requestClient.post(`/hr/development-plan/${id}/submit`);
}

/** 审核规划 */
export function reviewDevelopmentPlan(id: number, comment?: string) {
  return requestClient.post(`/hr/development-plan/${id}/review`, null, { params: { comment } });
}

/** 更新里程碑状态 */
export function updateMilestoneStatus(milestoneId: number, status: string, completionNote?: string) {
  return requestClient.post(`/hr/development-plan/milestones/${milestoneId}/status`, null, {
    params: { status, completionNote },
  });
}

